test/clients/azure/osa.go

```diff
@@ -0,0 +1,22 @@
+package azure
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// OSAResourceGroup returns the name of the resource group holding the OSA
+// cluster resources
+func (cli *Client) OSAResourceGroup(ctx context.Context, resourcegroup, name, location string) (string, error) {
+	appName := strings.Join([]string{"OS", resourcegroup, name, location}, "_")
+
+	app, err := cli.Applications.Get(ctx, appName, appName)
+	if err != nil {
+		return "", err
+	}
+
+	r, err := azure.ParseResourceID(*app.ManagedResourceGroupID)
```

Comment:
I don't think this is correct. This will fail with `Invalid Resource Id format`. Managed application resource ids are of the form `/subscriptions/<subscription_id>/resourceGroups/<resource_group>` and ParseResourceID is unable to handle that format.

Reply:
thanks @charlesakalugwu, I was too clever by half.